src/lxc/confile.c

```diff
@@ -487,6 +493,73 @@ static int set_config_net_veth_pair(const char *key, const char *value,
 	return network_ifname(netdev->priv.veth_attr.pair, value, sizeof(netdev->priv.veth_attr.pair));
 }
 
+static int set_config_net_veth_vlan_id(const char *key, const char *value,
+				  struct lxc_conf *lxc_conf, void *data)
+{
+	int ret;
+	struct lxc_netdev *netdev = data;
+
+	if (lxc_config_value_empty(value))
+		return clr_config_net_veth_vlan_id(key, lxc_conf, data);
+
+	if (!netdev)
+		return -1;
+
+	if (strcmp(value, "none") == 0) {
+		netdev->priv.veth_attr.vlan_id = BRIDGE_VLAN_NONE;
+	} else {
+		unsigned short vlan_id;
+		ret = get_u16(&vlan_id, value, 0);
+		if (ret < 0)
+			return -1;
+
+		if (vlan_id > BRIDGE_VLAN_ID_MAX)
+			return -1;
+
+		netdev->priv.veth_attr.vlan_id = vlan_id;
+	}
+
+	netdev->priv.veth_attr.vlan_id_set = true;
+	return 0;
+}
+
+static int set_config_net_veth_vlan_tagged_id(const char *key, const char *value,
+				       struct lxc_conf *lxc_conf, void *data)
+{
+	__do_free unsigned short *vlan_id = NULL;
+	__do_free struct lxc_list *list = NULL;
+	int ret;
+	struct lxc_netdev *netdev = data;
+
+	if (lxc_config_value_empty(value))
+		return clr_config_net_veth_vlan_tagged_id(key, lxc_conf, data);
+
+	if (!netdev)
+		return ret_set_errno(-1, EINVAL);
+
+	vlan_id = malloc(sizeof(*vlan_id));
+	if (!vlan_id)
+		return -1;
```

Comment:
I haven't thought about whether using a list is really ideal but it sure works for now. But you don't need the `malloc()`.  Add the following two helpers to the relevant other helpers in `src/lxc/macro.h`:
```c
#define UINT_TO_PTR(u) ((void *) ((uintptr_t) (u)))
#define PTR_TO_USHORT(p) ((unsigned short)(uintptr_t)(p)))
```
Then to store the pointer in the list:
```c
unsigned short vlan_id;
.
.
.
lxc_list_init(list)
list->elem = UINT_TO_PTR(vlan_id)
```
and to retrieve that `vlan_id` from the list:
```c
unsigned short vlan_id = PTR_TO_USHORT(list->elem)
```

Reply:
thanks @brauner. I've updated the PR with the use of these new helpers.

One thing I wanted to check with you thought, once I'd made these changes I also had to remove the `free(cur->elem);` line from https://github.com/lxc/lxc/pull/3435/files#diff-aa4fe548b9303c96febda542b6e2c92cR455 otherwise it segfaulted. Is this expected because we are no longer using malloc and the variable being pointed at by the pointer is already being cleaned up?

Reply:
Yes, nothing is allocated so you need to remove those `free()` calls. :)